src/Uno.UI.Composition/Composition/Visual.skia.cs

```diff
@@ -16,6 +15,49 @@ public partial class Visual : global::Microsoft.UI.Composition.CompositionObject
 	private CompositionClip? _clip;
 	private Vector2 _anchorPoint = Vector2.Zero; // Backing for scroll offsets
 	private int _zIndex;
+	private bool _matrixDirty = true;
+	private SKMatrix _totalMatrix = SKMatrix.Identity;
+
+	/// <returns>true if wasn't dirty</returns>
+	protected internal virtual bool SetMatrixDirty() => _matrixDirty = true;
+
+	/// <summary>
+	/// This is the final transformation matrix from the origin for this Visual.
+	/// </summary>
+	[DebuggerDisplay("{TotalMatrixString}")]
+	internal SKMatrix TotalMatrix
+	{
+		get
+		{
+			if (_matrixDirty)
+			{
+				_matrixDirty = false;
+
+				// Start out with the final matrix of the parent
+				var matrix = Parent?.TotalMatrix ?? SKMatrix.Identity;
+
+				// Set the position of the visual on the canvas (i.e. change coordinates system to the "XAML element" one)
+				var totalOffset = this.GetTotalOffset();
```

Comment:
Please use `Matrix3x2` or `Matrix4x4` as much as possible on convert it only once at the end to an `SKMatrix` (those are optimized to use CPU specific instruction set as much as possible)